tools/filelife.py

```diff
@@ -95,7 +99,8 @@
 TRACE_UNLINK_FUNC
 {
     struct data_t data = {};
-    u32 pid = bpf_get_current_pid_tgid() >> 32;
+    u64 pid_tgid = bpf_get_current_pid_tgid();
```

Comment:
Let us 'u32 tid = bpf_get_current_pid_tgid()' instead. Just search 'tid' in tools directory and you can find in most (if not all) cases, 'tid' is used as the key in the hash table between func entry and func return. Using 'tid' can be consistent with other tools. Otherwise the patch looks good. Indeed, we should check vfs_unlink return value to report unlink unless unlinking is really happening.

Reply:
Fix it , thanks, Song! and fix the following warning by ` birth.delete((struct dentry **)&data->dentry);`:

```
$ sudo ./filelife.py 
/virtual/main.c:113:18: warning: incompatible pointer types passing 'void **' to parameter of type 'struct dentry **' [-Wincompatible-pointer-types]
  113 |     birth.delete(&data->dentry);
      |                  ^~~~~~~~~~~~~
1 warning generated.
```